src/main/java6/net/finmath/marketdata/model/curves/ForwardCurveInterface.java

```diff
@@ -34,12 +41,15 @@
     double getForward(AnalyticModelInterface model, double fixingTime, double paymentOffset);
 
     /**
-	 * Returns the name of the discount curve associated with this forward curve.
-	 * 
-	 * @return The name of the discount curve associated with this forward curve.
-	 */
+     * @return The discountCurveName if forwardCurve is ForwardCurve, null otherwise
+     */
 	String getDiscountCurveName();
 	
+    /**
+     * @return The baseDiscountCurveName if forwardCurve is ForwardCurveFromDiscountCurve, null otherwise
+     */
+    String getBaseDiscountCurveName();
```

Comment:
It is not a good style to a method in a weaker type only for the case if the derived class is of type ForwardCurveFromDiscountCurve. This would force ALL other curve to implement this method. Why at all?

Also the name BaseDiscountCurve is maybe misleading. The word "base curve" might be associated with the financial concept of a base curve (where a curve is modelled above another curve, e.g. for Bond curve).

I would suggest removal of getBaseDisocuntCurveName from the interface.